src/tokenserver/db/models.rs

```diff
@@ -61,152 +70,294 @@ impl TokenserverDb {
-    }
+        let user = results::GetUser {
+            uid: raw_user.uid,
+            client_state: raw_user.client_state,
+            generation: raw_user.generation,
+            node: raw_user.node,
+            keys_changed_at: raw_user.keys_changed_at,
+            created_at: raw_user.created_at,
+            old_client_states,
+        };
 
-    #[cfg(test)]
-    fn post_node_sync(&self, node: params::PostNode) -> DbResult<results::PostNode> {
-        let query = r#"
-            INSERT INTO nodes (service, node, available, current_load, capacity, downed, backoff)
-            VALUES (?, ?, ?, ?, ?, ?, ?)
-        "#;
-        diesel::sql_query(query)
-            .bind::<Integer, _>(node.service_id)
-            .bind::<Text, _>(&node.node)
-            .bind::<Integer, _>(node.available)
-            .bind::<Integer, _>(node.current_load)
-            .bind::<Integer, _>(node.capacity)
-            .bind::<Integer, _>(node.downed)
-            .bind::<Integer, _>(node.backoff)
-            .execute(&self.inner.conn)?;
+        Ok(user)
+    }
 
-        let query = r#"
-            SELECT id
-              FROM nodes
+    /// Mark users matching the given email and service ID as replaced.
+    fn replace_users_sync(&self, params: params::ReplaceUsers) -> DbResult<results::ReplaceUsers> {
+        const QUERY: &str = r#"
+            UPDATE users
+               SET replaced_at = ?
              WHERE service = ?
-               AND node = ?
-               AND available = ?
-               AND current_load = ?
-               AND capacity = ?
-               AND downed = ?
-               AND backoff = ?
+               AND email = ?
+               AND replaced_at IS NULL
+               AND created_at < ?
         "#;
-        diesel::sql_query(query)
-            .bind::<Integer, _>(node.service_id)
-            .bind::<Text, _>(&node.node)
-            .bind::<Integer, _>(node.available)
-            .bind::<Integer, _>(node.current_load)
-            .bind::<Integer, _>(node.capacity)
-            .bind::<Integer, _>(node.downed)
-            .bind::<Integer, _>(node.backoff)
-            .get_result::<results::PostNode>(&self.inner.conn)
+        let timestamp = Self::get_timestamp_in_milliseconds();
+
+        diesel::sql_query(QUERY)
+            .bind::<Bigint, _>(timestamp)
+            .bind::<Integer, _>(&params.service_id)
+            .bind::<Text, _>(&params.email)
+            .bind::<Bigint, _>(timestamp)
+            .execute(&self.inner.conn)
+            .map(|_| ())
             .map_err(Into::into)
     }
 
-    #[cfg(test)]
-    fn post_service_sync(&self, service: params::PostService) -> DbResult<results::PostService> {
-        let query = r#"
-            INSERT INTO services (service, pattern)
-            VALUES (?, ?)
+    /// Mark the user with the given uid and service ID as being replaced.
+    fn replace_user_sync(&self, params: params::ReplaceUser) -> DbResult<results::ReplaceUser> {
```

Comment:
Analogous legacy Tokenserver DB method is [here](https://github.com/mozilla-services/tokenserver/blob/master/tokenserver/assignment/sqlnode/sql.py#L526)